### PR TITLE
org.blackfennec.app.extensions.core

### PR DESCRIPTION
**Pull request by SimonKindhauser**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.extensions.core

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.extensions.core

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.extensions.core

**Comment by SimonKindhauser** [comment]
The pipeline fails with the following error:  
`** (process:2893135): ERROR **: 17:26:32.680: Cannot get the default GSettingsSchemaSource - is the gsettings-desktop-schemas package installed?`

I don't know what could trigger it...

**Comment by dginovker** [comment]
You have to manually provide the pip dependencies like this

https://github.com/flathub/org.flathub.flatpak-external-data-checker/blob/master/org.flathub.flatpak-external-data-checker.yml#L271

https://github.com/flathub/org.flathub.flatpak-external-data-checker/blob/master/python3-jsonschema.json

You can read why and find a handy link here: https://github.com/flathub/flathub/wiki/App-Requirements#stable-releases-reproducible-builds. In particular:

> Accessing the network is not allowed during the build process, to download extra files, everything that an app requires must be included within the manifest. There is a [community provided set of tools](https://github.com/flatpak/flatpak-builder-tools) to make this easier for some projects (such as npm) that would traditionally want to access the network.

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.extensions.core

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.extensions.core

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.extensions.core

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.extensions.core

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.extensions.core

File: org.blackfennec.extensions.core.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+runtime-version: 'stable'
+sdk: org.gnome.Sdk//43
+build-extension: true
+modules:
+  - name: blackfennec.extensions.core
+    buildsystem: simple
+    build-commands:
+      # will be enabled with version 0.11
+      #- pip install --prefix=${FLATPAK_DEST} .
+      - install -Dm644 ./org.blackfennec.extensions.core.metainfo.xml ${FLATPAK_DEST}/share/metainfo/org.blackfennec.extensions.core.metainfo.xml
```
Comment (hfiguiere) [inline org.blackfennec.extensions.core.yml:13]: so right now this package just install the appstream file in `/app/share/metainfo`.

It doesn't look right.

Comment (SimonKindhauser) [reply org.blackfennec.extensions.core.yml:13]: we will uncomment line 12 in approx. 3 weeks for release 0.11. At the same time we will also remove line 29 and 30 of [org.balckfennec.app](https://github.com/flathub/org.blackfennec.app/blob/master/org.blackfennec.app.yml). The two lines are in conflict with each other.

File: org.blackfennec.extensions.core.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+id: org.blackfennec.extensions.core
```
Comment (barthalion) [inline org.blackfennec.extensions.core.yml:1]:
```suggestion
id: org.blackfennec.app.extensions.core
```

Extensions targeting single app should use that app's ID.

**Review by barthalion — COMMENTED** [review]
I don't think I really follow the idea here. Is the core extension expected to be released on a different cadence than the main app?

**Comment by SimonKindhauser** [comment]
> 

Not sure what you're asking..
The main app has been first released about 6 weeks ago (two sprints) and has so far included the core and base extensions.
With the next release in ~two weeks we would like to (finally) ship the extensions separate from the app. Generally speaking, the extensions will see updates once every sprint as does the main app.

The reason we have started early with the preparations (e.g. requesting the extensions to be available on flathub) is to ensure a seamless upgrade to our users. We would hate to ship a broken product. It would be a shame to remove the extensions from the app without them being available on flathub for a few days!

File: org.blackfennec.extensions.core.yml (hunk 2 above)
Comment (SimonKindhauser) [reply org.blackfennec.extensions.core.yml:1]: we would have named the app `org.blackfennec` if we could have :grinning:  
i personally prefer the version without the `app`. 

Is it a SHOULD or a COULD?
Consider it done iff its the former :smile:

**Comment by hfiguiere** [comment]
TBF I don't understand why you want to make these **flatpak** extensions. Even more since all is released at the same time and they seem to be required for the app to do anything.

**Comment by SimonKindhauser** [comment]
Duly noted. We hope for more useful extensions in the future too.

**Comment by barthalion** [comment]
> With the next release in ~two weeks we would like to (finally) ship the extensions separate from the app. Generally speaking, the extensions will see updates once every sprint as does the main app.

If the core extension is required to make the app work at all, and it's released at the same time when the app is updated, it doesn't make any sense to split it. The app has to work out of the box. The extensions should be used only for optional features.

File: org.blackfennec.extensions.core.yml (hunk 2 above)
Comment (barthalion) [reply org.blackfennec.extensions.core.yml:1]: Let me rephrase it: they need to match.

File: org.blackfennec.extensions.core.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+branch: '0.10'
+runtime: org.blackfennec.app
+runtime-version: 'stable'
+sdk: org.gnome.Sdk//43
+build-extension: true
+modules:
+  - name: blackfennec.extensions.core
```
Comment (SimonKindhauser) [inline org.blackfennec.extensions.core.yml:8]:
```suggestion
  - name: blackfennec.app.extensions.core
```

**Comment by SimonKindhauser** [comment]
The app works without it. However, our users expect it to be available.

In the future the release cadence of the main application and its extensions will differ.

Respectfully, the architecture of an application and thus its use of extensions should be up to the developers and maintainers. We do value your opinions but remain confident in our decisions.

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.app.extensions.core

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.app.extensions.core

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.app.extensions.core

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.app.extensions.core

**Comment by hfiguiere** [comment]
> The app works without it. However, our users expect it to be available.

Then it should come by default.

> In the future the release cadence of the main application and its extensions will differ.

That's not a problem. Just update that part in the manifest and roll a new build. It's like updating a dependency. And if your fear is that "user will have to redownload everything" I can tell you that you just missed the part were flatpak deduplicate files (via ostree) and thus the update would only download the changed files.
 
> Respectfully, the architecture of an application and thus its use of extensions should be up to the developers and maintainers. We do value your opinions but remain confident in our decisions.

Respectfully, when it come to the context of Flatpak, we are pretty confident this is the wrong approach. We are not arguing about your architecture, just that Flatpak extensions are definitely the wrong solution here. You'll get a lot of complaints about missing functionality, and if you default to *auto download* the extension, then there is no point in making it separate.

File: org.blackfennec.extensions.core.yml (hunk 2 above)
Comment (hfiguiere) [reply org.blackfennec.extensions.core.yml]: > we would have named the app `org.blackfennec` if we could have 😀
> i personally prefer the version without the `app`.

Thing is usually I outline that `app` isn't proper (I would have suggested `blackfennec` or anything that isn't `app` or `desktop`) but it appear that I didn't this time.

Comment (hfiguiere) [reply org.blackfennec.extensions.core.yml]: And just to clarify `org.blackfennec` would be rejected flat out.

**Comment by SimonKindhauser** [comment]
> Then it should come by default.

We will auto download it. The platform is designed such that even the core extension can be replaced by a user provided implementation.

> That's not a problem. [...]

We agree. This was in response to an earlier question by [barthalion](https://github.com/barthalion).

> You'll get a lot of complaints about missing functionality

Some things we will have to learn the hard way. We can always revert back to the old ways for the core module if it turns out to be a mistake.

> Respectfully, when it come to the context of Flatpak, we are pretty confident this is the wrong approach.

Hopefully, together we will find a solution we can all stand behind.

> and if you default to auto download the extension, then there is no point in making it separate.

This feature seems tailor made for our use case. It provides a sane default but allows advanced user to customize their experience if they choose to.

File: org.blackfennec.app.extensions.core.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+id: org.blackfennec.app.extensions.core
+branch: '0.10'
```
Comment (barthalion) [inline org.blackfennec.app.extensions.core.yml:2]: We discussed this briefly on Matrix, but I will repeat it here: you shouldn't use versioned branches for extensions targetting only apps. Once it's removed, I'm inclined to merge it, even though you will be switching back to the single manifest sooner than later.

Comment (SimonKindhauser) [reply org.blackfennec.app.extensions.core.yml:2]:
```suggestion
```

**Comment by SimonKindhauser** [comment]
bot, build org.blackfennec.app.extensions.core
